### PR TITLE
Fix clientId getter

### PR DESCRIPTION
**Pull request by helenye-stripe**
Fix for https://github.com/stripe/stripe-dotnet/issues/2902

### CODE REVIEW

File: src/Stripe.net/Stripe.net.csproj [hunk 1]
```diff
@@ -54,6 +54,10 @@
     <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
     <PackageReference Include="System.Configuration.ConfigurationManager" Version="7.0.0" />
   </ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework)' == 'net8.0'">
+    <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
+    <PackageReference Include="System.Configuration.ConfigurationManager" Version="8.0.0" />
+  </ItemGroup>
```
Comment (anniel-stripe) [inline src/Stripe.net/Stripe.net.csproj:60]: Is it safe to add this dependency?

File: src/StripeTests/Infrastructure/Public/StripeConfigurationTest.cs [hunk 2]
```diff
@@ -74,6 +74,32 @@ public void StripeClient_Getter_ThrowsIfClientIsNullAndApiKeyIsEmpty()
             }
         }
 
+        [Fact]
+        public void StripeClient_Getter_ThrowsIfClientIsNullAndClientIdIsEmpty()
```
Comment (anniel-stripe) [inline src/StripeTests/Infrastructure/Public/StripeConfigurationTest.cs:78]: We're not checking that it's throwing, we're just checking if it falls back to `System.Configuration.ConfigurationManager.AppSettings["StripeClientId"]` right? Maybe `StripeClient_Getter_UsesAppSettingsClientIdIfUnset`?

File: src/StripeTests/Infrastructure/Public/StripeConfigurationTest.cs [hunk 3]
```diff
@@ -74,6 +74,32 @@ public void StripeClient_Getter_ThrowsIfClientIsNullAndApiKeyIsEmpty()
...
+        {
+            var origApiKey = StripeConfiguration.ApiKey;
+            var origClientId = StripeConfiguration.ClientId;
+
+            try
+            {
+                StripeConfiguration.ApiKey = "sk_test_123";
```
Comment (anniel-stripe) [inline src/StripeTests/Infrastructure/Public/StripeConfigurationTest.cs:85]: Can we also check the same behavior for ApiKey while we're at it?

File: src/StripeTests/Infrastructure/Public/StripeConfigurationTest.cs [hunk 4]
```diff
@@ -74,6 +74,32 @@ public void StripeClient_Getter_ThrowsIfClientIsNullAndApiKeyIsEmpty()
...
+                StripeConfiguration.ClientId = string.Empty;
+                System.Configuration.ConfigurationManager.AppSettings["StripeClientId"] = "ca_stripeconfiguration";
+
+                StripeConfiguration.StripeClient = null;
+
+                var client = StripeConfiguration.StripeClient;
```
Comment (anniel-stripe) [inline src/StripeTests/Infrastructure/Public/StripeConfigurationTest.cs:91]: I don't know if this part is necessary?

File: src/Stripe.net/Stripe.net.csproj (hunk 1 above)
Comment (anniel-stripe) [reply src/Stripe.net/Stripe.net.csproj:60]: I think we can add to https://github.com/stripe/stripe-dotnet/blob/master/src/StripeTests/StripeTests.csproj

Comment (helenye-stripe) [reply src/Stripe.net/Stripe.net.csproj:60]: I think this should have already existed, looking at https://github.com/stripe/stripe-dotnet/pull/2650/files? Users can set this as well outside of tests no?

Comment (anniel-stripe) [reply src/Stripe.net/Stripe.net.csproj:60]: Oh OK I missed that

File: src/StripeTests/Infrastructure/Public/StripeConfigurationTest.cs [hunk 5]
```diff
@@ -1,6 +1,7 @@
 namespace StripeTests
 {
     using System;
+    using System.Configuration;
```
Comment (anniel-stripe) [inline src/StripeTests/Infrastructure/Public/StripeConfigurationTest.cs:4]:
```suggestion
```

File: src/Stripe.net/Stripe.net.csproj [hunk 6]
```diff
@@ -54,6 +54,10 @@
     <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
     <PackageReference Include="System.Configuration.ConfigurationManager" Version="7.0.0" />
   </ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework)' == 'net8.0'">
+    <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
+    <PackageReference Include="System.Configuration.ConfigurationManager" Version="8.0.0" />
```
[the remark this answers is not in the document]
Comment (helenye-stripe) [reply src/Stripe.net/Stripe.net.csproj]: I'll remove and open another PR

**Review by anniel-stripe — APPROVED** [review]

**Review by prathmesh-stripe — APPROVED** [review]

**Pull request merged**